### PR TITLE
Require Jenkins 2.346.3 or newer

### PR DESCRIPTION
**Pull request by MarkEWaite**
## Require Jenkins 2.346.3 or newer

- Require Jenkins 2.346.3 or newer
- Resolve character encoding warning in properties
- Convert from JUnit 3 HudsonTestCase to JUnit 4 JenkinsRule

Pull request is not complete, but may help others if they need to update this plugin to resolve implied dependencies.

Test failures are beyond my ability to resolve.

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by MarkEWaite** [comment]
Resolves the following pull requests by simplifying dependencies:

* #23
* #24 
* #27 

Supersedes the following dependency with a newer version: (can be reduced to an older version if use of most recent release is a concern):

* #26

**Comment by ikedam** [comment]
Thank you for the pull request. Some tests failed and I'll see and try to fix them.

**Comment by ikedam** [comment]
Relates: https://issues.jenkins.io/browse/JENKINS-70286

File: pom.xml [hunk 1]
```diff
@@ -38,90 +39,43 @@
...
-      <scope>test</scope>
+      <groupId>org.jenkins-ci.plugins</groupId>
+      <artifactId>matrix-auth</artifactId>
     </dependency>
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
-      <artifactId>cloudbees-folder</artifactId>
-      <version>4.0</version>
-      <scope>test</scope>
+      <artifactId>matrix-project</artifactId>
```
Comment (ikedam) [inline pom.xml:52]: Note: `MatrixProject` is used here: https://github.com/jenkinsci/jobcopy-builder-plugin/blob/jobcopy-builder-1.4.2/src/main/java/jp/ikedam/jenkins/plugins/jobcopy_builder/JobcopyBuilder.java#L354
and this cannot be optional.

**Comment by ikedam** [comment]
Note about change in this request: UTF-8 letters are no longer usable in properties files (even in comment lines!) as `InjectedTest` refuses that. (Relates https://github.com/jenkinsci/jenkins-test-harness/pull/433 ?)

**Comment by MarkEWaite** [comment]
> Note about change in this request: UTF-8 letters are no longer usable in properties files (even in comment lines!) as `InjectedTest` refuses that. (Relates [jenkinsci/jenkins-test-harness#433](https://github.com/jenkinsci/jenkins-test-harness/pull/433) ?)

I believe that if you want to use UTF-8 characters in property files, you need to require Java 11 and Jenkins 2.361.4 or newer.  The property files needed to be encoded ISO-8859-1 before the transition to Java 11.  Java 11 can read property files that are UTF-8 encoded.

**Comment by ikedam** [comment]
Test failures look related to this message:
```
WARNING hudson.model.ParametersAction#filter: Skipped parameter `fromJobName` as it is undefined on `testPerformWithAdditionalFileset2`. Set `-Dhudson.model.ParametersAction.keepUndefinedParameters=true` to allow undefined parameters to be injected as environment variables or `-Dhudson.model.ParametersAction.safeParameters=[comma-separated list]` to whitelist specific parameter names, even though it represents a security breach or `-Dhudson.model.ParametersAction.keepUndefinedParameters=false` to no longer show this message.
```

**Comment by ikedam** [comment]
Other tests (JobcopyBuilderPermissionTest) look fail for following:
```
WARNING hudson.model.Descriptor#save: Failed to save /workspace/target/tmp/j h13891172681329790620/jenkins.security.QueueItemAuthenticatorConfiguration.xml
java.lang.UnsupportedOperationException: Refusing to marshal org.springframework.security.authentication.UsernamePasswordAuthenticationToken for security reasons; see https://www.jenkins.io/redirect/class-filter/
```

**Comment by ikedam** [comment]
> JobcopyBuilderPermissionTest.testSystemSucceedOverwriteAsCreate
> JobcopyBuilderPermissionTest.testUserSucceedOverwriteAsCreate

They failed for the fix of Jenkins core SECURITY-321:
* https://www.jenkins.io/security/advisory/2017-02-01/#items-could-be-created-with-same-name-as-existing-item
* https://github.com/jenkinsci/jenkins/commit/d4f7fb1896f3b6a38eabb776e396a8c844c68adc

They should fail with Jenkins 2.32.2 and later. Expected results should be changed (I'll do that in following commits).

**Review by ikedam — APPROVED** [review]
All tests passed!
Thanks for the request.

**Pull request merged**